### PR TITLE
Add (filename #line) to log and only keep 20 log files

### PR DESCRIPTION
**Pull request by hu13**

### CODE REVIEW

File: pvHelpers/misc.py [hunk 1]
```diff
@@ -84,48 +100,103 @@ def updateLog(self, logobj):
         self.logobj = logobj
 
     def debug(self, string):
-        if self.logobj is not None:
-            self.logobj.debug(string)
-        else:
-            print "{} DEBUG: {}".format(self.__now(), string)
+        curframe = inspect.currentframe()
```
Comment (samantehrani) [inline pvHelpers/misc.py]: pls put these two lines under try as well. u could preset `currframe = calframe = None`

File: setup.py [hunk 2]
```diff
@@ -2,24 +2,18 @@
...
+    "protobuf==3.5.2.post1", "fipscrypto==1.0"
+]
 
 if sys.platform == "win32":
     install_requires += ["pywin32==224"]
 
 setup(
     name="pvHelpers",
-    version="4.60",
+    version="4.62",
```
Comment (samantehrani) [inline setup.py]: when release ready, let's use `70`. The intention being to use non-10-multiplies for dev/test ...

File: pvHelpers/misc.py [hunk 3]
```diff
@@ -84,48 +100,103 @@ def updateLog(self, logobj):
...
     def __now(self):
         return datetime.datetime.now().strftime("%Y/%m/%d %H:%M:%S")
 
+    def __format_caller_info(self, calframe):
+        """
+            :return: (filename, line_number, func)
+        """
+        filename = calframe[1][1]
+        line_number = calframe[1][2]
+        caller_func = calframe[1][3]
```
Comment (samantehrani) [inline pvHelpers/misc.py]: what's the behavior if there's no caller_func ?

Comment (hu13) [reply pvHelpers/misc.py]: added comment to clarify and also tests.

**Pull request merged**